fvm/crypto.go

```diff
@@ -97,11 +134,18 @@ func verifySignatureFromRuntime(
 	rawTag string,
 	message []byte,
 	rawPublicKey []byte,
-	rawSigAlgo string,
-	rawHashAlgo string,
+	signatureAlgorithm runtime.SignatureAlgorithm,
+	hashAlgorithm runtime.HashAlgorithm,
 ) (bool, error) {
-	sigAlgo := StringToSigningAlgorithm(rawSigAlgo)
-	hashAlgo := StringToHashingAlgorithm(rawHashAlgo)
+	sigAlgo := RuntimeToCryptoSigningAlgorithm(signatureAlgorithm)
+	if sigAlgo == crypto.UnknownSigningAlgorithm {
+		return false, fmt.Errorf("invalid signature algorithm: %s", signatureAlgorithm)
+	}
+
+	hashAlgo := RuntimeToCryptoHashingAlgorithm(hashAlgorithm)
+	if hashAlgo == hash.UnknownHashingAlgorithm {
+		return false, fmt.Errorf("invalid hash algorithm: %s", hashAlgorithm)
+	}
```

Comment:
Isn't it implicitly disabled then and the call `Verify` below (https://github.com/onflow/flow-go/pull/529/files#diff-f7f88b8bf4e799b156f025158a797c81b52b0874b07f942a4e6935b39643162cR162 ) will fail? I don't think we should add explicit checks of what is supported at call-sites (like here), but just return an error in the function that performs the actual logic (Verify below)

Reply:
I agree `BLS` would fail, as you mentioned, if it's used for Flow account keys. 
I am wondering about the case where a user writes a contract, uses BLS (since it's supported by Cadence) but is unable to use any hasher that goes with it. Is that scenario possible?  From a usability point of view (not security), it would be better to give users the full set of tools.